src/components/AutoComplete/AutoComplete.js

```diff
@@ -0,0 +1,40 @@
+import React, { memo, useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import checkProps from '@jam3/react-check-extra-props';
+
+import typingAhead from 'typing-ahead';
+
+const AutoComplete = ({
+  id,
+  data,
+  className,
+  autoCompleteResult
+}) => {
+  const [model, setModel] = useState(null);
+
+  function onInputChange({ target: { value } }) {
+    autoCompleteResult(value ? typingAhead.find(value, model) : null);
+  }
```

Comment:
Should this be implementing `useCallback()`?

Reply:
I think we should leave this up to the dev who uses the component to memoize the result of the `autoCompleteResult` function.